src/compiler/types.ts

```diff
@@ -676,6 +688,9 @@ namespace ts {
         elementType: TypeNode;
     }
 
+    export interface StructArrayTypeNode extends ArrayTypeNode {

```

Comment:
Consider branding this type with member `_structArrayTypeNodeBrand: void` to limit your ability to pass `ArrayTypeNode`s where `StructArrayTypeNode`s are expected.